File: Robot/app/src/main/java/com/changren/robot/RobotApplication.java
package com.changren.robot;

import android.app.Application;

import com.changren.robot.media.IMediaPlayer;
import com.changren.robot.media.MagicMediaPlayer;
import com.changren.robot.utils.XLog;

/**
 * Created by Admin
 * Date 2019/3/11.
 */

public class RobotApplication extends Application {

    private IMediaPlayer mPlayer;

    @Override
    public void onCreate() {
        super.onCreate();

        XLog.deBug(false);
    }

    public IMediaPlayer getPlayer(){
        if (mPlayer == null){
            mPlayer = new MagicMediaPlayer(this);
        }
        return mPlayer;
    }

    public IMediaPlayer stopPlayer(){
        if (mPlayer != null && mPlayer.isPlaying()){
            mPlayer.stop();
            mPlayer = null;
        }

        return mPlayer;
    }

    @Override
    public void onTerminate() {
        if (mPlayer != null){
            mPlayer.release();
        }
        super.onTerminate();
    }
}

File: Robot/appkotlin/src/main/java/com/changren/robot/RobotApplication.kt
package com.changren.robot

import android.app.Application
import com.changren.robot.media.IMediaPlayer
import com.changren.robot.media.MagicMediaPlayer
import com.changren.robot.utils.XLog

/**
 * Created by Admin
 * Date 2019/3/29.
 */
class RobotApplication : Application() {

    private var mPlayer: IMediaPlayer? = null

    override fun onCreate() {
        super.onCreate()
        XLog.deBug(true)
    }

    val player: IMediaPlayer
        get(){
            if (mPlayer == null) {
                mPlayer = MagicMediaPlayer(this)
            }
            return mPlayer as IMediaPlayer
        }

    fun stopPlayer(): IMediaPlayer? {
        if (mPlayer != null && mPlayer!!.isPlaying) {
            mPlayer!!.stop()
            mPlayer = null
        }
        return mPlayer
    }

    override fun onTerminate() {
        if (mPlayer != null){
            mPlayer!!.release()
        }
        super.onTerminate()
    }
}
